sample_apps/simple_app_python/example_app.py

```diff
@@ -202,55 +198,52 @@ def do_main(args) -> bool:
             sys.exit(1)
 
     return result
 
 ###############################################################################
-def write_cert_pvt_key_to_files(cctm, whoami, data_dir, print_all):
+def write_certficates_to_file(cctm, whoami, data_dir, print_all):
     """
-    Persist certificate(s) and private-keys that have been established as
+    Persist app's and root-certificate(s) that have been established as
     part of certification. Data from cc_trust_manager{} is written to disk,
     so that, later, Python interfaces requiring certificates can use this
     data from file(s).
     """
-    assert whoami == 'server'
+    assert whoami in ('client', 'server')
 
     print('cc_auth_key_initialized_ = ', cctm.cc_auth_key_initialized_)
     print('primary_admissions_cert_valid_ = ', cctm.primary_admissions_cert_valid_)
 
     cert_outfile = os.path.join(data_dir, APPLN_CERT_SIGNED_BY_ROOT_CA)
-    pkey_outfile = os.path.join(data_dir, APPLN_KEYF_SIGNED_BY_ROOT_CA)
-
-    ca_root_outfile = os.path.join(data_dir, ROOT_CA_POLICY_CERT)
     dump_cert_to_file(cctm.serialized_primary_admissions_cert_,
-                      cert_outfile, print_all)
+                      cert_outfile, whoami, print_all)
 
-    dump_private_key_to_file(cctm, pkey_outfile, print_all)
-
-    dump_cert_to_file(cctm.serialized_policy_cert_, ca_root_outfile, print_all)
+    ca_root_outfile = os.path.join(data_dir, ROOT_CA_POLICY_CERT)
+    dump_cert_to_file(cctm.serialized_policy_cert_, ca_root_outfile, whoami, print_all)
 
 ###############################################################################
-def server_dispatch(data_dir, server_app_host, server_app_port):
+def server_dispatch(cctm, data_dir:str, server_app_host:str, server_app_port:int,
+                    print_all: bool):
     """
     Start a server process listening on a SSL socket waiting for client input.
-    """
-    print(fnl(), 'Start server process ...\n')
 
-    # Server needs to authenticate the client; hence 'Purpose.CLIENT_AUTH'
-    context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
-    context.verify_mode = ssl.CERT_REQUIRED
-
-    cert_outfilename = os.path.join(data_dir, APPLN_CERT_SIGNED_BY_ROOT_CA)
-    key_outfilename  = os.path.join(data_dir, APPLN_KEYF_SIGNED_BY_ROOT_CA)
-    context.load_cert_chain(certfile = cert_outfilename, keyfile = key_outfilename)
+    Parameters:
+      cctm      - Certifier Trust manager object
+      data_dir  - Path of provisioning data directory, where certificate
+                  file(s) has been persisted earlier
+      server_app_host
+      server_app_port
+                - Server app's hostname and port number
+    """
+    print('\n**** ', fnl(), 'Start server process ...\n')
 
-    ca_root_outfilename = os.path.join(data_dir, ROOT_CA_POLICY_CERT)
-    context.load_verify_locations(cafile = ca_root_outfilename)
+    ssl_ctxt = setup_server_ssl_context(cctm, data_dir, print_all)
```

Comment:
This modularization is modeled on what was done for the pytest case, `test_server_process_with_mtls_root_signed_cert_using_temp_pvt_key_file()`, which also calls a similar `setup_server_ssl_context()` method.